includes/class-amp-reader-themes.php

```diff
@@ -0,0 +1,512 @@
+				return in_array( $theme->slug, $supported_themes, true );
+			}
+		);
+
+		return $supported_themes_from_response;
+	}
+
+	/**
+	 * Prepares a single theme.
+	 *
+	 * @param array|object $theme Theme data from the wordpress.org themes API.
+	 * @return array|object Prepared theme array.
+	 */
+	public function prepare_theme( $theme ) {
+		$theme_array = (array) $theme;
+
+		$keys = [
+			'name',
+			'slug',
+			'preview_url',
+			'screenshot_url',
+			'homepage',
+			'description',
+			'requires',
+			'requires_php',
+			'download_link',
+		];
+
+		$prepared_theme = array_filter(
+			$theme_array,
+			static function ( $key ) use ( $keys ) {
+				return in_array( $key, $keys, true );
+			},
+			ARRAY_FILTER_USE_KEY
+		);
+
+		foreach ( $keys as $key ) {
+			if ( ! array_key_exists( $key, $prepared_theme ) ) {
+				$prepared_theme[ $key ] = '';
+			}
+		}
+
+		return $prepared_theme;
+	}
+
+	/**
+	 * Provides the current theme name.
+	 *
+	 * @return string|bool The theme name, or false if the theme has errors.
+	 */
+	private function get_current_theme_name() {
+		if ( null === $this->current_theme_name ) {
+			$current_theme = wp_get_theme();
+
+			$this->current_theme_name = $current_theme->exists() ? $current_theme->get( 'Name' ) : false;
+		}
+
+		return $this->current_theme_name;
+	}
+
+	/**
+	 * Returns whether a theme can be installed on the system.
+	 *
+	 * @param array $theme Theme data.
+	 * @return bool True if themes can be installed.
+	 */
+	public function can_install_theme( $theme ) {
+		// @todo Add support for installing non-default reader themes. Until that is done, themes that are provided via
+		// the amp_reader_themes filter will show on the reader themes screen but will need to be manually installed on
+		// the site.
+		$default_reader_theme_slugs = wp_list_pluck( self::DEFAULT_READER_THEMES, 'slug' );
+		if ( ! in_array( $theme['slug'], $default_reader_theme_slugs, true ) ) {
+			return false;
+		}
+
+		if ( null === $this->can_install_themes ) {
+			if ( ! class_exists( 'WP_Upgrader' ) ) {
+				require_once ABSPATH . 'wp-admin/includes/file.php';
+				require_once ABSPATH . 'wp-admin/includes/class-wp-upgrader.php';
+			}
+
+			$this->can_install_themes = true === ( new WP_Upgrader() )->fs_connect( [ get_theme_root() ] );
+		}
+
+		if ( ! $this->can_install_themes ) {
+			return false;
+		}
+
+		if ( ! empty( $theme['requires'] ) && version_compare( get_bloginfo( 'version' ), $theme['requires'], '<' ) ) {
+			return false;
+		}
+
+		if ( ! empty( $theme['requires_php'] ) && version_compare( phpversion(), $theme['requires_php'], '<' ) ) {
+			return false;
+		}
+
+		return true;
+	}
+
+	/**
+	 * Returns reader theme availability status.
+	 *
+	 * @param array $theme Theme data.
+	 * @return string Theme availability status.
+	 */
+	public function get_theme_availability( $theme ) {
+		switch ( true ) {
+			case $this->get_current_theme_name() === $theme['name']:
```

Comment:
Instead of using the `name`, this should really be using the `slug`, no?

Reply:
As such, this could simply be replaced with:

```suggestion
			case get_template() === $theme['slug']:
```

Reply:
Good call.